src/main/java/mate/academy/spring/dao/impl/UserDaoImpl.java

```diff
@@ -0,0 +1,50 @@
+package mate.academy.spring.dao.impl;
+
+import java.util.List;
+import mate.academy.spring.dao.UserDao;
+import mate.academy.spring.model.User;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class UserDaoImpl implements UserDao {
+    private final SessionFactory sessionFactory;
+
+    @Autowired
+    public UserDaoImpl(SessionFactory sessionFactory) {
+        this.sessionFactory = sessionFactory;
+    }
+
+    @Override
+    public void add(User user) {
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.save(user);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Can't add user to DB", e);

```

Comment:
```suggestion
            throw new RuntimeException("Can't add user to DB " + user, e);

```
 don't forget about informative exception messages

Reply:
done